src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1050,10 +1053,27 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
             }
 
             EmitBlock(catchBlock.Body);
-
             _builder.CloseLocalScope();
         }
 
+        private static bool IsFilterConstantFalse(BoundCatchBlock catchBlock)
+        {
+            // Depending on compilation options, BoundLiteral can be wrapped by BoundSequence or BoundSequencePointExpression
+            if(catchBlock.ExceptionFilterOpt != null)
+            {
+                if(catchBlock.ExceptionFilterOpt is BoundSequence sequence)
+                {
+                    return sequence.Value is BoundSequencePointExpression expression
```

Comment:
Use of `expression` in different scopes is confusing.
@gafter, I don't remember exactly the scoping rules for is-pattern variable declarations. Does this seem right?